bin/test-clouds

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+#
+# Run integration tests on 4 cloud providers:
+# - Amazon (EKS)
+# - DigitalOcean (DO)
+# - Google (GKE)
+# - Microsoft (AKS)
+#
+# This script assumes you have a working Kubernetes cluster set up on each Cloud
+# provider, and that Kubernetes contexts are configured via environment
+# variables.
+
+set -eu
+
+# TODO: share this with test-run
+function check_linkerd_binary(){
+    printf "Checking the linkerd binary..."
+    if [[ "$linkerd_path" != /* ]]; then
+        printf "\\n[%s] is not an absolute path\\n" "$linkerd_path"
+        exit 1
+    fi
+    if [ ! -x "$linkerd_path" ]; then
+        printf "\\n[%s] does not exist or is not executable\\n" "$linkerd_path"
+        exit 1
+    fi
+    exit_code=0
+    "$linkerd_path" version --client > /dev/null 2>&1 || exit_code=$?
+    if [ $exit_code -ne 0 ]; then
+        printf "\\nFailed to run linkerd version command\\n"
+        exit $exit_code
+    fi
+    printf "[ok]\\n"
+}
+
+linkerd_path=$1
+check_linkerd_binary
+
+printf "\nKicking off tests for:\n- $AKS\n- $DO\n- $EKS\n- $GKE\n\n"
```

Comment:
should we consider running a subset of these if the user only has some of them set up? or is this mostly for internal testing, so we can assume all of them will be set up?

Reply:
yeah, this is mostly for internal testing. i'm inclined to leave it as-is. if a user wanted to test their own changes against a subset of cloud providers, i'd leave it up to them manually edit this script to disable certain providers.